node-bootstrapper/main.go

```diff
@@ -60,24 +63,36 @@ func (s SensitiveString) UnsafeValue() string {
 	return string(s)
 }
 
+func isWindows() bool {
+	return os.PathSeparator == '\\' && os.PathListSeparator == ';'
+}
+
+func mainWithDefer(logFile *os.File) error {
+	defer logFile.Close()
+
+	logger := slog.New(slog.NewJSONHandler(logFile, nil))
+	slog.SetDefault(logger)
+
+	slog.Info("node-bootstrapper started")
+	ctx := context.Background()
+	return Run(ctx)
+}
+
 func main() {
-	logFile, err := os.OpenFile(LogFile, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
+	logFile, err := getLogFile()
 	if err != nil {
 		//nolint:forbidigo // there is no other way to communicate the error
 		fmt.Printf("failed to open log file: %s\n", err)
 		os.Exit(1)
 	}
-	defer logFile.Close()
 
-	logger := slog.New(slog.NewJSONHandler(logFile, nil))
-	slog.SetDefault(logger)
+	// mainWithDefer is responsible for closing the log file. That way we know it's closed
+	// and can use os.Exit safely in this method.
+	err = mainWithDefer(logFile)
```

Comment:
you can't use logger past this point.
I usually try to put creating and closing things in the same function to avoid such issues.

I think I sorted it out differently in the main.

Reply:
Yup - your solution is to not use a defer and to just close it manually. Which also works. So I'll not migrate the "runWithLog" code over :)